kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/foodproducers/ExtraFoodShopProducer.java

```diff
@@ -0,0 +1,41 @@
+package com.kodilla.good.patterns.challenges.foodproducers;
+
+public class ExtraFoodShopProducer implements ProducerProcess {
+
+    private String name;
+    private String residenceAddress;
+
+    public ExtraFoodShopProducer(String name, String residenceAddress) {
+        this.name = name;
+        this.residenceAddress = residenceAddress;
+    }
+
+    @Override
+    public boolean process(User user, Product product, StoreService storeService) {
```

Comment:
You provide dependency as an argument - I think it is not the best idea - definitely is unusual. It is strange that Processor which has StoreService pass it to Producer - Producer can do check if product is available on its own!
And now you have duplication of code, because every producer do the same check.